src/scatter_buildblock/ScatterEstimation.cxx

```diff
@@ -0,0 +1,1486 @@
+                %this->tail_mask_par_filename);
+        //return Succeeded::no;
+        create_tail_mask_from_acfs.ACF_threshold = 1.1;
+        create_tail_mask_from_acfs.safety_margin = 4;
+    }
+
+    create_tail_mask_from_acfs.set_input_projdata_sptr(mask_projdata);
+    create_tail_mask_from_acfs.set_output_projdata_sptr(this->mask_projdata_sptr);
+    return create_tail_mask_from_acfs.process_data();
+}
+
+//! If the filters are not applied in this specific order the
+//! results are not the desirable every time.
+bool
+ScatterEstimation::
+apply_mask_in_place(DiscretisedDensity<3, float>& arg,
+                    const mask_parameters& _this_mask)
+{
+    // Re-reading every time should not be a problem, as it is
+    // just a small txt file.
+    PostFiltering<DiscretisedDensity<3, float> > filter;
+
+    if(filter.parse(this->mask_postfilter_filename.c_str()) == false)
+    {
+        warning(boost::format("Error parsing postfilter parameters file %1%. Aborting.")
+                %this->mask_postfilter_filename);
+        return false;
+    }
+
+    //1. add_scalar//2. mult_scalar//3. power//4. min_threshold//5. max_threshold
+
+    pow_times_add pow_times_thres_max(0.0f, 1.0f, 1.0f, NumericInfo<float>().min_value(),
+                                      0.001f);
+    pow_times_add pow_times_add_scalar( -0.00099f, 1.0f, 1.0f, NumericInfo<float>().min_value(),
+                                        NumericInfo<float>().max_value());
+
+    pow_times_add pow_times_thres_min(0.0, 1.0f, 1.0f, 0.0f,
+                                      NumericInfo<float>().max_value());
+
+    pow_times_add pow_times_times( 0.0f, 100002.0f, 1.0f, NumericInfo<float>().min_value(),
+                                   NumericInfo<float>().max_value());
+
+    pow_times_add pow_times_add_object(_this_mask.add_scalar,
+                                       _this_mask.times_scalar,
+                                       1.0f,
+                                       _this_mask.min_threshold,
+                                       _this_mask.max_threshold);
+    // 1. filter the image
+    filter.process_data(arg);
+
+    // 2. max threshold
+    in_place_apply_function(arg,
+                            pow_times_thres_max);
+    // 3. add scalar
+    in_place_apply_function(arg,
+                            pow_times_add_scalar);
+    // 4. min threshold
+    in_place_apply_function(arg,
+                            pow_times_thres_min);
+    // 5. times scalar
+    in_place_apply_function(arg,
+                            pow_times_times);
+    // 6. Add 1.
+    in_place_apply_function(arg,
+                            pow_times_add_object);
+
+    return true;
+}
+
+int ScatterEstimation::get_iterations_num() const
+{
+    return num_scatter_iterations;
+}
+
+Succeeded
+ScatterEstimation::prepare_projdata(const shared_ptr<ProjData> input,
+                                    shared_ptr<ProjData> output)
+{
+
```

Comment:
what is this function supposed to do?

Reply:
To simplify the set_up(s). But then I saw that although I keep doing almost the same things on different projdata thee similarity is not that good, and dropped it. I forgot to delete it.